diff --git ../demo/src/main/java/org/jboss/jdf/examples/view/PerformanceBean.java ../demo/src/main/java/org/jboss/jdf/examples/view/PerformanceBean.java
index 689cc8c..f9aa6c1 100644
--- ../demo/src/main/java/org/jboss/jdf/examples/view/PerformanceBean.java
+++ ../demo/src/main/java/org/jboss/jdf/examples/view/PerformanceBean.java
@@ -25,6 +25,8 @@ import javax.persistence.criteria.Predicate;
 import javax.persistence.criteria.Root;

 import org.jboss.jdf.example.ticketmonster.model.Performance;
+import org.jboss.jdf.example.ticketmonster.model.Section;
+import org.jboss.jdf.example.ticketmonster.model.SectionAllocation;
 import org.jboss.jdf.example.ticketmonster.model.Show;

 /**
@@ -122,8 +124,13 @@ public class PerformanceBean implements Serializable
       {
          if (this.id == null)
          {
-            this.entityManager.persist(this.performance);
-            return "search?faces-redirect=true";
+				this.entityManager.persist(this.performance);
+				for (Section s : this.performance.getShow().getVenue()
+						.getSections()) {
+					this.entityManager.persist(new SectionAllocation(
+							this.performance, s));
+				}
+				return "search?faces-redirect=true";
          }
          else
          {
diff --git ../demo/src/main/webapp/WEB-INF/web.xml ../demo/src/main/webapp/WEB-INF/web.xml
index d1afd7b..e62b752 100644
--- ../demo/src/main/webapp/WEB-INF/web.xml
+++ ../demo/src/main/webapp/WEB-INF/web.xml
@@ -32,7 +32,4 @@
   <session-config>
     <session-timeout>30</session-timeout>
   </session-config>
-  <welcome-file-list>
-    <welcome-file>faces/index.xhtml</welcome-file>
-  </welcome-file-list>
 </web-app>
diff --git ../demo/src/main/webapp/admin/index.html ../demo/src/main/webapp/admin/index.html
index 902113f..4036f32 100644
--- ../demo/src/main/webapp/admin/index.html
+++ ../demo/src/main/webapp/admin/index.html
@@ -1,70 +1 @@
-<!DOCTYPE html>
-<html>
-<head>
-    <title>Ticket Monster</title>
-    <meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
-    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=0;">
-
-
-    <link type="text/css" rel="stylesheet" href="../resources/css/screen.css"/>
-    <link rel="stylesheet" href="../resources/css/bootstrap.css" type="text/css" media="all"/>
-    <link rel="stylesheet" href="../resources/css/custom.css" type="text/css" media="all">
-
-    <link href='http://fonts.googleapis.com/css?family=Rokkitt' rel='stylesheet' type='text/css'>
-
-    <!-- Add JavaScript library for IE6-8 support of HTML5 elements -->
-    <!--[if lt IE 9]>
-    <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
-    <![endif]-->
-
-</head>
-<body>
-
-<!--
-    Helper page, to be replaced by Forge generation
--->
-<div id="logo"><div class="wrap"><h1>Ticket Monster</h1></div></div>
-<div id="container">
-    <div id="menu">
-        <div class="navbar">
-            <div class="navbar-inner">
-                <div class="container">
-                    <ul class="nav">
-                        <li><a href="../#about">About</a></li>
-                        <li><a href="../#events">Events</a></li>
-                        <li><a href="../#venues">Venues</a></li>
-                        <li><a href="../#bookings">Bookings</a></li>
-                        <li><a href="../booking-monitor.html">Monitor</a></li>
-                        <li><a href="../admin">Administration</a></li>
-                    </ul>
-                </div>
-            </div>
-        </div>
-    </div>
-    <div id="content" class="container-fluid">
-        <div class="well">
-            <h2 class="page-header special-title">Don't Forget About Forge</h2>
-
-            <p>You must use JBoss Forge to generate the administration console. For doing so,
-            follow the next steps:</p>
-
-            <ol>
-                <li>Install JBoss Forge</li>
-                <li>Start JBoss Forge by running the following in the command prompt: <pre>$ forge</pre></li>
-                <li>Verify that the Forge plugin is installed by running <pre>$ forge list-plugins</pre>  and verifying that <code>org.richfaces.forge.richfaces-forge-plugin</code> is in the returned list.</li>
-                <li>If the outcome of the previous step was that the Richfaces plugin was not installed, do that now: <pre>$ forge install-plugin richfaces</pre></li>
-                <li>Once the Forge prompt is up, run: <pre>$ run admin_layer.fsh</pre> (answering <i>yes</i> on questions concerning patches)</li>
-                <li>Rebuild and redeploy</li>
-            </ol>
-
-            <p>Navigate back to this page, you should see the Forge landing page instead.</p>
-        </div>
-    </div>
-</div>
-
-<footer style="">
-    <div style="text-align: center;"><img src="../resources/img/dualbrand_as7eap.png" alt="HTML5"/></div>
-</footer>
-
-</body>
-</html>
\ No newline at end of file
+<meta http-equiv="refresh" content="0;url=../faces/admin/index.xhtml" />
\ No newline at end of file
diff --git ../demo/src/main/webapp/admin/index.xhtml ../demo/src/main/webapp/admin/index.xhtml
new file mode 100644
index 0000000..844aa67
--- /dev/null
+++ ../demo/src/main/webapp/admin/index.xhtml
@@ -0,0 +1,30 @@
+<?xml version='1.0' encoding='UTF-8' ?>
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
+<ui:composition xmlns="http://www.w3.org/1999/xhtml"
+	xmlns:ui="http://java.sun.com/jsf/facelets"
+	template="/resources/scaffold/pageTemplate.xhtml">
+
+	<ui:param name="pageTitle" value="Welcome to Forge" />
+
+	<ui:define name="header">
+		Welcome to Forge
+	</ui:define>
+
+	<ui:define name="main">
+		<h2 class="success">
+			Your application is running.
+		</h2>
+		<p>
+			<a target="_blank"
+				href="http://forge.jboss.org/docs/index.html">Documentation</a>
+			| <a target="_blank"
+				href="http://twitter.com/intent/tweet?text=I+just+built+and+deployed+a+webapp+using+%23Forge,+from+%23JBoss+-+Super+easy!+(+http://jboss.org/forge+)">Get
+				Excited!</a><br /> <a target="_blank" href="http://jboss.org/forge"><br />
+				Forge Project</a> | <a target="_blank"
+				href="https://community.jboss.org/en/forge">User
+				Forums</a> | <a target="_blank"
+				href="https://issues.jboss.org/browse/FORGE">Report an issue</a>
+		</p>
+	</ui:define>
+
+</ui:composition>
\ No newline at end of file
diff --git ../demo/src/main/webapp/admin/event/view.xhtml ../demo/src/main/webapp/admin/event/view.xhtml
index 451479f..b952db0 100644
--- ../demo/src/main/webapp/admin/event/view.xhtml
+++ ../demo/src/main/webapp/admin/event/view.xhtml
@@ -4,6 +4,7 @@
 	xmlns:h="http://java.sun.com/jsf/html"
 	xmlns:f="http://java.sun.com/jsf/core"
 	xmlns:ui="http://java.sun.com/jsf/facelets"
+    xmlns:tm="http://java.sun.com/jsf/composite/tm"
 	template="/resources/scaffold/pageTemplate.xhtml">

 	<f:metadata>
@@ -31,6 +32,7 @@
 			<h:outputLabel for="eventBeanEventMediaItem" value="Media Item:"/>
 			<h:link id="eventBeanEventMediaItem" outcome="/admin/mediaItem/view" value="#{eventBean.event.mediaItem}">
 				<f:param name="id" value="#{eventBean.event.mediaItem.id}"/>
+                <tm:image media="#{mediaManager.getPath(eventBean.event.mediaItem)}" />
 			</h:link>
 			<h:outputText/>
 			<h:outputLabel for="eventBeanEventCategory" value="Category:"/>
diff --git ../demo/src/main/webapp/admin/venue/view.xhtml ../demo/src/main/webapp/admin/venue/view.xhtml
index 06edc33..6676fbf 100644
--- ../demo/src/main/webapp/admin/venue/view.xhtml
+++ ../demo/src/main/webapp/admin/venue/view.xhtml
@@ -4,6 +4,7 @@
 	xmlns:h="http://java.sun.com/jsf/html"
 	xmlns:f="http://java.sun.com/jsf/core"
 	xmlns:ui="http://java.sun.com/jsf/facelets"
+    xmlns:tm="http://java.sun.com/jsf/composite/tm"
 	xmlns:forgeview="http://jboss.org/forge/view"
 	template="/resources/scaffold/pageTemplate.xhtml">

@@ -44,7 +45,8 @@
 			<h:outputText/>
 			<h:outputLabel for="venueBeanVenueMediaItem" value="Media Item:"/>
 			<h:link id="venueBeanVenueMediaItem" outcome="/admin/mediaItem/view" value="#{venueBean.venue.mediaItem}">
-				<f:param name="id" value="#{venueBean.venue.mediaItem.id}"/>
+                <tm:image media="#{mediaManager.getPath(venueBean.venue.mediaItem)}" />
+                <f:param name="id" value="#{venueBean.venue.mediaItem.id}"/>
 			</h:link>
 			<h:outputText/>
 			<h:outputLabel for="venueBeanVenueDescription" value="Description:"/>
diff --git ../demo/src/main/webapp/index.html ../demo/src/main/webapp/index.html
index b1c9bee..ba1b595 100644
--- ../demo/src/main/webapp/index.html
+++ ../demo/src/main/webapp/index.html
@@ -1 +1,36 @@
-<meta http-equiv="refresh" content="0;url=./faces/index.xhtml" /> 
\ No newline at end of file
+<!DOCTYPE html>
+<html>
+<head>
+    <title>Ticket Monster</title>
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
+    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=0"/>
+
+    <script type="text/javascript" src="resources/js/libs/modernizr-2.0.6.js"></script>
+
+    <!--
+         A simple check on the client. For touch devices or small-resolution screens)
+         show the mobile client. By enabling the mobile client on a small-resolution screen
+         we allow for testing outside a mobile device (like for example the Mobile Browser
+         simulator in JBoss Tools and JBoss Developer Studio).
+      -->
+    <script type="text/javascript">
+        var loader = document.createElement("script")
+        loader.setAttribute("src", "resources/js/libs/require.js");
+        if ( navigator.userAgent == "TicktetMonster Cordova Webview iOS" ) {
+            environment="hybrid-ios"
+        }
+        else if ( navigator.userAgent == "TicketMonster Cordova Webview Android" ) {
+            environment="hybrid-android"
+        }
+        else if (Modernizr.touch || Modernizr.mq("only all and (max-width: 480px)")) {
+            environment = "mobile"
+        } else {
+            environment = "desktop"
+        };
+        loader.setAttribute("data-main","resources/js/configurations/" + environment);
+        document.head.appendChild(loader)
+    </script>
+</head>
+<body>
+</body>
+</html>
\ No newline at end of file
diff --git ../demo/src/main/webapp/resources/tm/image.xhtml ../demo/src/main/webapp/resources/tm/image.xhtml
new file mode 100644
index 0000000..ceafffb
--- /dev/null
+++ ../demo/src/main/webapp/resources/tm/image.xhtml
@@ -0,0 +1,25 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
+        "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
+<html xmlns="http://www.w3.org/1999/xhtml"
+      xmlns:h="http://java.sun.com/jsf/html"
+      xmlns:composite="http://java.sun.com/jsf/composite">
+<head>
+    <title>Cached Image</title>
+</head>
+<body>
+
+<composite:interface>
+    <composite:attribute name="media"
+                         type="org.jboss.jdf.example.ticketmonster.service.MediaPath" />
+    <composite:attribute name="id" type="java.lang.String" />
+</composite:interface>
+
+<composite:implementation>
+    <h:graphicImage value="#{cc.attrs.media.url}"
+                    rendered="#{!cc.attrs.media.cached}" />
+    <h:graphicImage value="/rest/media/cache/#{cc.attrs.media.url}"
+                    rendered="#{cc.attrs.media.cached}" />
+</composite:implementation>
+
+</body>
+</html>
\ No newline at end of file
